src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -174,9 +178,21 @@ private constructor(
     if (messages.lastOrNull()?.id == message.id) {
       messages.removeLast()
     }
-    messages.add(message)
-    chatPanel.addOrUpdateMessage(message)
-    HistoryService.getInstance().update(internalId, messages)
+    if (messages.size == 0) {
+      selectedModel = chatPanel.modelDropdown.selectedItem?.toString() ?: ""
+      sessionId.get().thenAccept { sessionId ->
+        CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+          agent.server.webviewReceiveMessage(WebviewReceiveMessageParams(sessionId, WebviewMessage(command = "chatModel", model = selectedModel) )).get()
+          messages.add(message)
+          chatPanel.addOrUpdateMessage(message, shouldAddBlinkingCursor = true)
+          HistoryService.getInstance().update(internalId, messages, selectedModel)
+        }
+      }
+    } else {
+      messages.add(message)
+      chatPanel.addOrUpdateMessage(message, shouldAddBlinkingCursor = true)
+      HistoryService.getInstance().update(internalId, messages)
+    }
```

Comment:
I think it shouldn't be possible for user to send any message without a selected model (from logical perspective). Is this branching (if-else) really necessary here?

Perhaps we can generalize this problem? This fragment is IMO smell: `selectedModel = chatPanel.modelDropdown.selectedItem?.toString() ?: ""`. Why not `... ?: "anthropic/claude-2.0"`?

Reply:
For the second part i agree but regarding the first part: this if/esle branch is here because we need to set model only when first message is sent. For all other messages we need to use already selected model.